lib/schema/deviceEvent.js

```diff
@@ -196,7 +196,19 @@ module.exports = function(streamDAO){
             payload: schema.ifExists(schema.isObject)
           }
         }
+      ),
+      schema.makeHandler(
+        'pumpSettingsOverride',
+        {
+          schema: {
+            deviceTime: schema.validDeviceTime,
+            overrideType: schema.in('physicalActivity', 'sleep'),
+            method: schema.ifExists(schema.in('automatic', 'manual')),
+            duration: schema.and(schema.isNumber, schema.greaterThanEq(0), schema.lessThanEq(604800000)),
+            payload: schema.ifExists(schema.isObject)
+          }
```

Comment:
Do we need the ability to update this (via `previous`)? What about the currently active `pumpSettingsOverride` at the moment of upload?

Reply:
I suppose that implies we need an `expectedDuration`. Did we miss that in the data model updates document/review?

Reply:
Instead of writing a whole bunch of new Jellyfish code to update durations via `previous`, I'm doing the following: 
- If `pumpSettingsOverride` is active at time of upload, extend duration until the end of the upload
- On the next upload, have the first `pumpSettingsOverride` event (where we only know the time it ended) start at the time of the first event. It gets annotated with `'tandem/pumpSettingsOverride/estimated-duration`. This same logic is used if it's the first event in the pump history (i.e., not a delta upload)
- Since there are CGM events every five minutes when CIQ is active, there will be a maximum 5-minute gap between the two `pumpSettingsOverride` events.

When we switch Tandem from Jellyfish to platform, we can add the proper logic in Uploader to update the previous event. I've added a TODO in the code, see https://github.com/tidepool-org/uploader/pull/1346/files#diff-30c72e8dac23589016b2a5e7b55d45410e1092a4b7892cff0fcaab98f2fd7e6eR332-R358

With regards to `expectedDuration`: Since we don't have the sleep schedules, we don't know the `expectedDuration` in the event that a sleep activity gets cancelled. Exercise activities don't have a programmed duration, so I don't see us using `expectedDuration` for those either?

Reply:
Okay, that makes sense. Also, leaving out `expectedDuration` makes sense. (We did add `expectedDuration` to the platform data model updates.) Should be all good. 👍